### ISSUE TITLE
JSON schema OneOf data matching issue in array.

### ISSUE THREAD

**Issue by dev-kanishk** [ngx-formly/ngx-formly#3805]
**Description**
When using oneOf in an array, formly only matches data of first element of array.
In the following schema:
```
{
 "type": "object",
 "properties": {
  "headings": {
   "title": "Headings",
   "type": "array",
   "items" : {
    "oneOf" : [
     {
      "title" : "Standard Heading", 
      "oneOf" : [
       {
        "title": "Heading 1",
        "const": "heading-1"
       },
       {
        "title": "Heading 2",
        "const": "heading-2"
       }
      ]
     },
     {
      "title" : "Custom Heading", 
      "type" : "string" 
     }
    ]
   }
  }
 }
}
```

And the data:

```
{
  "options": [
    "random",
    "Heading 1"
  ]
}
```

The second field of the array is rendered as string input "custom string" rather it should have matched the Enum value "Heading 1"

**Minimal Reproduction**
https://stackblitz.com/edit/angular-kuxmzz-o9gbon?file=src%2Fapp%2Fapp.component.ts

**Your Environment**
- Formly version: 6.2.2

**Comment by dev-kanishk** [comment]
I looked into the codebase in formly-json-schema.service.ts, we are calling the isFieldValid method to know if the field is valid. Here in case of array, we are building the Enum field only once for the first element of array. Thus for the second element the isFieldValid returns the validity of first array field.

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.3.0](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.3.0) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
